### PR TITLE
feat(leadspace-with-search): update to v2

### PR DESCRIPTION
**Pull request by IgnacioBecerra**
### Related Ticket(s)
[#10787](https://github.com/carbon-design-system/carbon-for-ibm-dotcom/issues/10787)

### Description
**Note: We have to merge the Leadspace v2 PR first**

This PR introduces many changes to the Leadspace with Search component. 
The greatest change is the removal of many of its sub-components, now just using `leadspace-heading` and `leadspace-with-search-copy` for its text content.

### Changelog

**New**

- leadspace with search also supports the heading highlight 

**Changed**

- cleaned up the scss styles 

**Removed**

- Removed `<dds-leadspace-with-search-heading>`, `<dds-leadspace-with-search-content>`, 
`<dds-leadspace-with-search-content-heading>`, `<dds-leadspace-with-search-content-copy>`
- No more image leadspace with search variant

### CODE REVIEW

**Comment by oliviaflory** [comment]
On Friday Sep 30, @IgnacioBecerra and I discussed simplifying the `Adjacent theme` knob.

Currently we offer five adjacent theme options.
Screenshot 2023-10-02 at 11 53 43 AM
and the adopter can also change the page theme (or Carbon theme in storybook) which causes some conflicts and criss crossing of theming tokens, resulting in blue 60 text on gray 90 and mixing dark and light theme backgrounds in some cases.
Screenshot 2023-10-02 at 11 59 36 AM
Screenshot 2023-10-02 at 12 00 08 PM



In **v11**, we have the ability to **inline theme** so we only need to provide two adjacent theme options
- Mono theme
- Dual theme
And the adopters can then choose if the lead space theme, which will result in all the correct color values, and the remaining page can be whatever they choose, even white.

**Review by oliviaflory — APPROVED** [review]
@IgnacioBecerra LOVE IT!

File: docs/dotcom-v2-migration.md [hunk 1]
```diff
@@ -47,8 +47,9 @@ For Carbon v11 migration guidance, see their
 | footer                    | No API changes.                                                                                                                                |
 | hr                        | View changes [here](#horizontal-rule)                                                                                                          |
 | image                     | View changes [here](#image)                                                                                                                    |
-| image-with-caption        | This component is deprecated in v1 and has been removed in v2 in favor of the image component                                                  |
+| image-with-caption        | Replaced by `image`                                                                                                                            |
```
Comment (ariellalgilmore) [inline docs/dotcom-v2-migration.md:50]: just curious if there is a reason for the change above? Trying to keep the docs consistent

File: packages/styles/scss/components/search-with-typeahead/_search-with-typeahead.scss [hunk 2]
```diff
@@ -206,208 +206,214 @@
...
-        &::after {
-          content: none;
-        }
-      }
+    .#{$prefix}--search-magnifier-icon {
+      position: absolute;
+      z-index: 2;
+      top: 50%;
+      left: $spacing-05;
+      fill: var(--cds-icon-secondary, #525252);
```
Comment (ariellalgilmore) [inline packages/styles/scss/components/search-with-typeahead/_search-with-typeahead.scss:313]:
```suggestion
      fill: $icon-secondary;
```

File: packages/styles/scss/components/search-with-typeahead/_search-with-typeahead.scss [hunk 3]
```diff
@@ -206,208 +206,214 @@
...
-      top: $spacing-11;
+    .#{$prefix}--header__search--close {
+      @include breakpoint(md) {
+        height: $spacing-09;
+      }
     }
 
-    &[theme-sticky] {
-      top: $spacing-05;
+    .react-autosuggest__suggestions-container {
```
Comment (ariellalgilmore) [inline packages/styles/scss/components/search-with-typeahead/_search-with-typeahead.scss:376]: maybe we can remove this react specific style?

**Review by ariellalgilmore — COMMENTED** [review]
Looking goood!!! 🎉  I think just the `leadspace-with-search` codesandbox examples need to be updated too

**Review by ariellalgilmore — APPROVED** [review]
LGTM!! just run a yarn format to get the ci-checks going: https://github.com/carbon-design-system/carbon-for-ibm-dotcom/actions/runs/6458468487/job/17532251119?pr=10961

**Pull request merged**